### PR TITLE
Retry connection on height mismatch

### PR DESCRIPTION
**Pull request by roman-khimov**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ Changelog for NeoFS Node
 - Double voting for validators on IR startup (#2365)
 - Skip unexpected notary events on notary request parsing step (#2315)
 - Session inactivity on object PUT request relay (#2460)
+- Missing connection retries on IR node startup whe the first configured mainner RPC node is not in sync (#XXX)
```
Comment (carpawell) [inline CHANGELOG.md:19]:
```suggestion
- Missing connection retries on IR node startup when the first configured mainnet RPC node is not in sync (#2474)
```

File: pkg/morph/subscriber/subscriber.go [hunk 2]
```diff
@@ -68,6 +68,7 @@ func (s *subscriber) NotificationChannels() NotificationChannels {
 }
 
 var (
+	ErrStaleNode = errors.New("RPC node is not yet up to date")
```
Comment (carpawell) [inline pkg/morph/subscriber/subscriber.go:71]: public err with no comments. not critical but usually we do not do that

**Review by carpawell — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (roman-khimov) [reply CHANGELOG.md]: Fixed.

File: pkg/morph/subscriber/subscriber.go (hunk 2 above)
Comment (roman-khimov) [reply pkg/morph/subscriber/subscriber.go:73]: Fixed.

File: CHANGELOG.md (hunk 1 above)
Comment (carpawell) [reply CHANGELOG.md]: but what about "mainner"? isn't it "mainnet"?

Comment (roman-khimov) [reply CHANGELOG.md]: Ouch. Fixed.

Comment (carpawell) [reply CHANGELOG.md]: 🥺

Comment (roman-khimov) [reply CHANGELOG.md]: Maybe now?

Comment (roman-khimov) [reply CHANGELOG.md]: Maybe now?

**Review by carpawell — APPROVED** [review]

**Review by cthulhu-rider — APPROVED** [review]

**Pull request merged**